test/coreSpec.js

```diff
@@ -21,32 +21,43 @@ var validOptions = {
     }
 };
 
-describe('Granim Core: ', function(){
-    beforeEach(function(){
+describe('Granim Core: ', function () {
+    beforeEach(function () {
         unsetCanvas();
         spyOn(console, 'error');
     });
 
-    describe('Invalid options: ', function(){
-        var func = function(){ return new Granim({element: '#granim-canvas'})};
+    describe('Invalid options: ', function () {
+        var func = function () { return new Granim({ element: '#granim-canvas' }) };
 
-        it('throws error on invalid element id ', function(){
+        it('throws error on invalid element id ', function () {
             expect(func).toThrowError("`#granim-canvas` could not be found in the DOM");
         });
 
-        it('throws an error when passed in anything other than a HTMLCanvasElement as an element', function(){
+        it('throws an error when passed in anything other than a HTMLCanvasElement as an element', function () {
             var invalidOptions = {
                 element: document.createElement('div'),
                 name: validOptions.name,
                 opactiy: validOptions.opactiy,
                 states: validOptions.states
             };
-            var func = function(){ return new Granim(invalidOptions)};
+            var func = function () { return new Granim(invalidOptions) };
             expect(func).toThrowError("The element you used is neither a String, nor a HTMLCanvasElement");
         });
+        it('throws an error when the option direction  has the value "custom" but not the customDirection object when a new Granim instance is created', function () {
+            var invalidOptions = {
+                element: validOptions.element,
+                name: validOptions.name,
+                opactiy: validOptions.opactiy,
+                states: validOptions.states,
+                direction: 'custom'
+            };
+            var func = function () { return new Granim(invalidOptions) };
+            expect(func).toThrowError('When using a custom direction, the custom object is required with it\'s attributes: x0, x1, y0, y1 of type number');
```

Comment:
`expect(func).toThrowError('Granim: Input error on customDirection option.\nCheck the API https://sarcadass.github.io/granim.js/api.html.');`